lib/syskit/data_service.rb

```diff
@@ -90,10 +90,13 @@ def find_device_attached_to(service = nil)
                     elsif driver_services.size > 1
                         raise ArgumentError, "#{self} handles more than one device, you must specify one of #{driver_services.map(&:name).sort.join(", ")} explicitely"
                     end
-                    service = driver_services.first
+                    service = driver_services.find do |driver_service|
+                        service == driver_service
+                    end
+                    
                 end
 
-                arguments["#{service.name}_dev"]
```

Comment:
Again, a small code example:
```
JointControl::FakeJointDriverTask.driver_for BesmanTest::FakeDevices::JointsDriver, :as => 'joints1'
JointControl::FakeJointDriverTask.driver_for BesmanTest::FakeDevices::JointsDriver, :as => 'joints2'
```
```
device(BesmanTest::FakeDevices::JointsDriver, :as => 'left_arm') \
                .with_conf('same_conf').prefer_deployed_tasks("driver_deployment")
            device(BesmanTest::FakeDevices::JointsDriver, :as => 'right_arm') \
                .with_conf('sample_conf').prefer_deployed_tasks("driver_deployment")
```
This example should be valid and result in selecting "driver_deployment" for both devices? I was not aware of this possiblity.

If the feature is needed and used, of cause we need to support it. I'll have a look at your proposed method with the default argument.